crates/adapters/src/controller/stats.rs

```diff
@@ -1607,14 +1640,12 @@ impl OutputEndpointStatus {
         endpoint_name: &str,
         config: &OutputEndpointConfig,
         total_processed_records: u64,
+        initial_statistics: Option<&CheckpointOutputEndpointMetrics>,
     ) -> Self {
         Self {
             endpoint_name: endpoint_name.to_string(),
             config: config.clone(),
-            metrics: OutputEndpointMetrics {
-                total_processed_input_records: AtomicU64::new(total_processed_records),
-                ..Default::default()
-            },
+            metrics: OutputEndpointMetrics::new(total_processed_records, initial_statistics),
             fatal_error: Mutex::new(None),
```

Comment:
I wonder if we should checkpoint the fatal error too.

Reply:
> I wonder if we should checkpoint the fatal error too.

I'm not sure.

In favor of checkpointing it is that we preserve the message.

Against it, though, is that we've restarted the connector from where it left off and one can expect that it will either succeed now (maybe the secrets have been fixed, for example), in which case we don't want that message to persist, or it will fail again in either the same way or a new way and in that case we do want the new fatal error message.

For now, I didn't add anything.